src/main/java/com/google/sps/servlets/NewClass.java

```diff
@@ -44,18 +49,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       String className = request.getParameter("className").trim();
 
-      String idToken = request.getParameter("idToken");
-      FirebaseToken decodedToken = authInstance.verifyIdToken(idToken);
+      Query query = new Query("Class");
+      query.addProjection(new PropertyProjection("name", String.class));
```

Comment:
A more efficient way to do this query would be to add a `FilterPredicate` to the query checking that the names are equal and just check the length of the result with `countEntities()`. Then you wouldn't have to iterate over every class. 

https://cloud.google.com/appengine/docs/standard/java/datastore/retrieving-query-results